source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -212,8 +212,10 @@ void StreamHandleWrapper::onSuccess(Http::MessagePtr&& response) {
   response->headers().iterate(
       [](const Http::HeaderEntry& header, void* context) -> Http::HeaderMap::Iterate {
         lua_State* state = static_cast<lua_State*>(context);
-        lua_pushstring(state, header.key().c_str());
-        lua_pushstring(state, header.value().c_str());
+        std::string header_key_string(header.key().getStringView());
```

Comment:
One random comment here: In the current header map code we are still adding a null byte at the end of each string. One follow up here would be to get rid of that null byte append which would allow 1 more byte to fit it inline space, and provide a small speed boost (and also make this string conversion required vs. just using `data()` which technically would work in the current code I think).

Reply:
Yeah, this occurred to me too. I'll create a separate issue for this follow up. I definitely want that to be a narrow and carefully reviewed change, and only once we've eliminated all of the other TODOs.

Reply:
https://github.com/envoyproxy/envoy/issues/6581 for this.

Reply:
(and I agree `data()` would work here, but given future desires to remove the NUL byte, it seems like a really risky short term thing)

Reply:
I just checked and you should be able to use `lua_pushlstring` here and elsewhere to avoid the string copy.